### PR TITLE
test(pkg): demo files not being copied in empty builds

### PR DESCRIPTION
**Pull request by Alizter**
Reproduction case for #9017

### CODE REVIEW

**Comment by Alizter** [comment]
I have a fix for this ready, though I am not 100% I understand why it works.

File: test/blackbox-tests/test-cases/pkg/opam-package-with-files-install.t [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  $ solve foo
+  Solution for dune.lock:
+  - foo.0.0.1
+
+The build action copying files/ is missing.
+  $ cat dune.lock/foo.pkg
+  (version 0.0.1)
+
+The foo.install file in files/ should have been copied over.
+  $ build_pkg foo 2> /dev/null
```
Comment (rgrinberg) [inline test/blackbox-tests/test-cases/pkg/opam-package-with-files-install.t:21]: Don't think this step is necessary.

File: test/blackbox-tests/test-cases/pkg/opam-package-with-files-install.t [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+  $ mkpkg foo
+  $ mkdir -p $mock_packages/foo/foo.0.0.1/files
+  $ cat > $mock_packages/foo/foo.0.0.1/files/foo.install
+
+  $ solve foo
+  Solution for dune.lock:
+  - foo.0.0.1
+
+The build action copying files/ is missing.
```
Comment (rgrinberg) [inline test/blackbox-tests/test-cases/pkg/opam-package-with-files-install.t:16]: There is no "build action" for copying files/

Comment (Alizter) [reply test/blackbox-tests/test-cases/pkg/opam-package-with-files-install.t]: You're right, I've rewritten it to something clearer now.

File: test/blackbox-tests/test-cases/pkg/opam-package-with-files-install.t (hunk 1 above)
Comment (Alizter) [reply test/blackbox-tests/test-cases/pkg/opam-package-with-files-install.t:24]: I've added a check for the .files directory. You can now see that it is in the lock file.

**Comment by rgrinberg** [comment]
The solving step wasn't necessary so I got rid of it.

Thanks for the PR.

**Pull request merged**